framework/wazuh/core/config/models/server.py

```diff
@@ -0,0 +1,333 @@
+    receiving_file : PositiveInt
+        The timeout for receiving files in seconds. Default is 120.
+    """
+    cluster_request: PositiveInt = 20
+    dapi_request: PositiveInt = 200
+    receiving_file: PositiveInt = 120
+
+
+class CommunicationsConfig(BaseModel):
+    """Configuration for communications settings.
+
+    Parameters
+    ----------
+    zip : ZipConfig
+        Configuration for zip settings.
+    timeouts : CommunicationsTimeoutConfig
+        Configuration for communication timeouts.
+    """
+    zip: ZipConfig = ZipConfig()
+    timeouts: CommunicationsTimeoutConfig = CommunicationsTimeoutConfig()
+
+
+class WorkerIntervalsConfig(BaseModel):
+    """Configuration for worker intervals.
+
+    Parameters
+    ----------
+    sync_integrity : PositiveInt
+        The interval for synchronizing integrity. Default is 9.
+    keep_alive : PositiveInt
+        The interval for sending keep-alive signals. Default is 60.
+    connection_retry : PositiveInt
+        The number of retries for connection attempts. Default is 10.
+    """
+    sync_integrity: PositiveInt = 9
+    keep_alive: PositiveInt = 60
+    connection_retry: PositiveInt = 10
+
+
+class WorkerRetriesConfig(BaseModel):
+    """Configuration for worker retries.
+
+    Parameters
+    ----------
+    max_failed_keepalive_attempts : PositiveInt
+        The maximum number of failed keep-alive attempts before considering the worker dead. Default is 2.
+    """
+    max_failed_keepalive_attempts: PositiveInt = 2
+
+
+class WorkerConfig(BaseModel):
+    """Configuration for worker nodes.
+
+    Parameters
+    ----------
+    intervals : WorkerIntervalsConfig
+        Configuration for worker intervals.
+    retries : WorkerRetriesConfig
+        Configuration for worker retries.
+    """
+    intervals: WorkerIntervalsConfig = WorkerIntervalsConfig()
+    retries: WorkerRetriesConfig = WorkerRetriesConfig()
+
+
+class SharedFiles(BaseModel):
+    """Configuration for shared files.
+
+    Parameters
+    ----------
+    dir : str
+        The directory containing shared files.
+    description : str
+        A description of the shared files.
+    permissions : PositiveInt
+        The permissions for the shared files.
+    source : str
+        The source of the shared files.
+    names : List[str]
+        The names of the shared files.
+    recursive : bool
+        Whether to search for files recursively.
+    restart : bool
+        Whether to restart the service when these files change.
+    remove_subdirs_if_empty : bool
+        Whether to remove subdirectories if they are empty.
+    extra_valid : bool
+        Whether to perform extra validation.
+    """
+    dir: str
+    description: str
+    permissions: PositiveInt
+    source: str
+    names: List[str]
+    recursive: bool
+    restart: bool
+    remove_subdirs_if_empty: bool
+    extra_valid: bool
+
+
+class ServerInternalConfig(BaseModel):
```

Comment:
I would call it something like `ServerSyncConfig` instead, `internal` is too generic and doesn't say much about its use

Reply:
Changed in e4e54327c2b3a45cbe77a6f9c30cfd19dc9ce0b6.